sdk/src/wallet/operations/announce_candidacy.rs

```diff
@@ -17,10 +17,7 @@ where
     crate::client::Error: From<S::Error>,
 {
     /// Announce a staking account's candidacy for the staking period.
-    pub async fn announce_candidacy(
-        &self,
-        account_id: impl Into<Option<AccountId>> + Send,
-    ) -> crate::wallet::Result<BlockId> {
+    pub async fn announce_candidacy(&self, account_id: AccountId) -> crate::wallet::Result<BlockId> {
         self.submit_basic_block(Payload::CandidacyAnnouncement(CandidacyAnnouncementPayload), account_id)
```

Comment:
I feel like we should add a check here to see if the account has enough BIC for this block without going negative and return an error if it's not enough, better error here than getting the account blocked

Reply:
can do!

Reply:
done!

Reply:
Thanks!